.github/workflows/build-deploy.yml

```
@@ -26,74 +30,147 @@ jobs:
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
 
-      - name: Configure AWS credentials
-        uses: aws-actions/configure-aws-credentials@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2.1.0
         with:
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-region: ${{ secrets.AWS_REGION }}
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Login to Amazon ECR
-        id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
+      # This step outputs Docker labels and Docker tag associated with github.ref (branch name in this scenario)
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4.3.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
-      - name: Build & push docker images
-        env:
-          BUILD_ID: ${{needs.unique_id.outputs.unique_id}}
-          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+      # When running on develop branch we want to tag Docker images with unique id and branch name only.
+      # When running on master and release we're adding sem ver and latest (for release) tags too.
+      - name: Setup Docker tags
+        id: tags
         run: |
-          docker build --build-arg SENTRY_RELEASE="${{needs.unique_id.outputs.unique_id}}" -t $ECR_REGISTRY/ics-api:$BUILD_ID -f docker/prod/Dockerfile .
-          docker push $ECR_REGISTRY/ics-api:$BUILD_ID
-          docker build --build-arg PUBLIC_IMAGE=$ECR_REGISTRY/ics-api:$BUILD_ID --build-arg SENTRY_COMMIT="$GITHUB_SHA" --build-arg SENTRY_RELEASE="${{needs.unique_id.outputs.unique_id}}" --build-arg SENTRY_AUTH_TOKEN=${{ secrets.SENTRY_AUTH_TOKEN }} -f docker/sentry-upload/Dockerfile .
-      - name: Logout of Amazon ECR
-        if: always()
-        run: docker logout ${{ steps.login-ecr.outputs.registry }}
+              if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
+                echo "TAGS=${{ steps.meta.outputs.tags }},${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{  needs.configure.outputs.IMAGE_TAG  }}" >> $GITHUB_OUTPUT
+              else
+                echo "TAGS=${{ steps.meta.outputs.tags }},${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{  needs.configure.outputs.IMAGE_TAG  }},${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{  needs.tag-release.outputs.tag  }}" >> $GITHUB_OUTPUT
+              fi
+              
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: ${{  needs.configure.outputs.PUSH_DOCKER  }}
+          platforms: linux/amd64,linux/arm64
```

I don't think our image is runnable on arm64 as I don't think we are building with arm as a target. Though I could be wrong about this.
```suggestion
          platforms: linux/amd64
```

---

Hmm, maybe support for `arm64` will indeed work. We can give it a try.